src/main/java/de/tum/in/www1/artemis/service/dto/athena/TextFeedbackDTO.java

```diff
@@ -24,18 +26,23 @@ public record TextFeedbackDTO(long id, long exerciseId, long submissionId, Strin
     public static TextFeedbackDTO of(long exerciseId, long submissionId, @NotNull Feedback feedback, TextBlock feedbackBlock) {
         Integer startIndex = feedbackBlock == null ? null : feedbackBlock.getStartIndex();
         Integer endIndex = feedbackBlock == null ? null : feedbackBlock.getEndIndex();
-        return new TextFeedbackDTO(feedback.getId(), exerciseId, submissionId, feedback.getText(), feedback.getDetailText(), feedback.getCredits(), startIndex, endIndex);
+        var gradingInstructionId = feedback.getGradingInstruction() == null ? null : feedback.getGradingInstruction().getId();
+        return new TextFeedbackDTO(feedback.getId(), exerciseId, submissionId, feedback.getText(), feedback.getDetailText(), feedback.getCredits(), gradingInstructionId,
+                startIndex, endIndex);
     }
 
     /**
      * Creates a TextBlockRef (feedback + text block combined) from this DTO and a TextSubmission
      */
-    public TextBlockRef toTextBlockRef(TextSubmission onSubmission) {
+    public TextBlockRef toTextBlockRef(TextSubmission onSubmission, GradingInstruction gradingInstruction) {
         Feedback feedback = new Feedback();
         feedback.setId(id());
-        feedback.setText(text());
-        feedback.setDetailText(detailText());
+        feedback.setText(title());
+        feedback.setDetailText(description());
         feedback.setCredits(credits());
+        // The given grading instruction should match the one of this DTO:
+        assert gradingInstructionId() == null || gradingInstructionId().equals(gradingInstruction.getId());
```

Comment:
We should not use assert in production code. Java will just spike these instructions if `-ea` is not set, which is typically not the case in production environments.